### PR TITLE
[Homework #17] - Class-and-functional-components by Edelmiro Antón

### PR DESCRIPTION
**Pull request by EdelmiroAnton**
Hi @ilya-ischenko!
Here's my hw. Let me know if there's something to improve.

Thanks!

### CODE REVIEW

**Review by ilya-ischenko — COMMENTED** [review]
Functionality looks good.
Need to update UI according to design

**Comment by EdelmiroAnton** [comment]
Hey @ilya-ischenko! I've just added the styles like the template. 

Let me know if I have to fix something else.

Thanks!

File: homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  addAvatar = async () => {
+    this.fetchAvatar().then((avatar) => {
+      this.setState({
+        avatar: [...this.state.avatar, { ...avatar }],
+      });
+    });
+    // console.log(...this.state.avatar);
+  };
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/App.js:30]: Read this one, please - https://en.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous
It is not valid to use this.state inside setState

File: homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const resp = await fetch('https://tinyfac.es/api/data?limit=1&quality=0');
+    const data = await resp.json();
+    // console.log(data);
+    return data[0];
+  };
+
+  addAvatar = async () => {
+    this.fetchAvatar().then((avatar) => {
+      this.setState({
+        avatar: [...this.state.avatar, { ...avatar }],
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/App.js:26]: Do you need both "async" and "then"?

File: homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/AddButton.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+export default class AddButton extends React.Component {
+  render() {
+    return (
+      <>
+        <div className="addButton" onClick={this.props.onClick}>
+          <div className="verticalLine"></div>
+          <div className="horizontalLine"></div>
+        </div>
+        ;
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/17-class-and-functional-components/my-avatar/src/components/AddButton.js:13]: Typo

**Comment by EdelmiroAnton** [comment]
Hey @ilya-ischenko! I've just fix and improve the code with your suggestions. Also delete unnecessary code.

Let me know if it's ok.

Thanks in advance!

**Pull request merged**